libs/product/src/models/product.ts

```diff
@@ -17,7 +18,7 @@ export enum DaffProductTypeEnum {
 /**
  * An interface for a product usable by the @daffodil/product library.
  */
-export interface DaffProduct extends DaffLocatable, DaffIdentifiable {
+export interface DaffProduct extends DaffLocatable, DaffIdentifiable, Partial<DaffCanonicallyLocatable> {
```

Comment:
Why is this a partial?

Reply:
Canonical URLs aren't needed (I assume) for product previews so it should be optional.